### PR TITLE
Feature/rework ttt

### PR DESCRIPTION
**Pull request by yenda**
This enables TTT on ropsten with the following contract https://ropsten.etherscan.io/address/0xc61aa0287247a0398589a66fcd6146ec0f295432#contracts

# Tests
We tested the following features with @siphiuel on Android and iOS simulator on ropsten network:

* set tribute
* remove tribute
* not able to send messages to a user with a set tribute
* pay tribute to a user and be able to send messages to that user

status: ready

### CODE REVIEW

**Review by vitvly — APPROVED** [review]

File: src/status_im/data_store/contacts.cljs [hunk 1]
```diff
@@ -1,10 +1,13 @@
 (ns status-im.data-store.contacts
   (:require [re-frame.core :as re-frame]
+            [cljs.tools.reader.edn :as edn]
+            [taoensso.timbre :as log]
             [status-im.data-store.realm.core :as core]))
 
 (defn- deserialize-contact [contact]
   (-> contact
       (update :tags #(into #{} %))
+      (update :tribute-to-talk edn/read-string)
```
Comment (flexsurfer) [inline src/status_im/data_store/contacts.cljs]: we use transit for account settings

**Review by flexsurfer — APPROVED** [review]

**Review by cammellos — CHANGES_REQUESTED** [review]
Please add some tests
@siphiuel @flexsurfer please do not approve large PRs with at least a bare minimum test coverage

**Comment by yenda** [comment]
@annadanchenko @churik please proceed with testing writing tests won't impact anything

**Comment by asemiankevich** [comment]
we are working on this @yenda :)

**Comment by yenda** [comment]
@cammellos tests added please review again

**Comment by churik** [comment]
@yenda testing is still in progress, so far:

#### 1) If TTT is set on both account in chat, only one of them should pay
Is it expected? I'm not sure
**Steps:**
- create 2 accounts (A and B)
- change network to Ropsten
- as user A: set TTT
- as user B: set TTT
- as user A: open profile of user B
- as user A: pay TTT and send message to user A
Currently after this User A and User B can chat regardless that User B didn't pay TTT for user A.

#### 2) User is not added in contacts, but TTT says that user is added to contacts after paying TTT
download

#### 3) Buttons are overlapped on some phones
Screenshot at May 31 18-08-11
Device: huawei p10 lite android8 5.2»

#### 4) No warning that TTT is set when user sends a message from a different network

**Steps:**
- create 2 accounts (A and B)
- as user A: change network to Ropsten
- as user A:  set TTT
- as user B: send message to user A (on mainnet)
**Expected result:**  can see that need to change network to contact user A, message sending is not possible
**Actual result:** message is sent, but in fact, it is not received and it will never be received.

#### 5) Warning with TTT enabled is shown on wallet screen
I don't think that this is expected
photo_2019-05-31 18 39 22


#### 6) On message TTT is paid no timestamp (design)
@errorists can you elaborate why on message "Tribute is paid" no timestamp is shown?
IMG_52B290F60830-1

#### 7) @rachelhamlin it will be breaking change for previous builds, right?
So now: if user from the previous build sends a message to user with TTT enabled, it just will be never received and both sides will not be notified about it.

**Comment by yenda** [comment]
@churik 

1. yes that is normal, only one side needs to pay for the conversation to start
2. we could improve the copy here but user is not added in contacts so showing the add to contact here is correct
3. will have a look
4. this cannot be done, that's the way it is (could be doable with swarm, but not with on-chain contract)
5. wrong picture?
6. it's not a message it's state
7. yes they will never receive the message

I'm going to make a build with ttt disabled please test it to see if we can merge this, we will handle the fixes separately

**Comment by churik** [comment]
@yenda changed the image for 5, mistake - sorry

**Comment by yenda** [comment]
@churik ok I made one more PR with ttt enabled to test fixes for 2 3 and 5

**Comment by yenda** [comment]
once it is built I will make a final build without ttt enabled

**Comment by yenda** [comment]
@churik 
Build 32: ttt enabled on ropsten
Build 33: ttt disabled

**Review by cammellos — APPROVED** [review]

**Comment by churik** [comment]
@yenda thanks for fixes: verified all. New issues:

### on build 32:
#### 8. Can't get messages after paying TTT
**Steps:**
- create 2 accounts (A and B)
- change network to Ropsten
- as user A: set TTT
- as user B: pay TTT and send message to user A
**Expected result:** user A will receive a message
**Actual result:** user A doesn't receive messages.
Ao basically chat flow is not working on build 32.

#### 9. "Tribute paid" message is always visible when open chat regardless of the message order
**Steps:**
- create 2 accounts (A and B)
- change network to Ropsten
- as user A: set TTT
- as user B: pay TTT and send message to user A
- as user B: send 10+ messages to user A
- go back and open chat with user A again

**Expected result:** can see last 10 messages, "Tribute paid" message is not shown until I will scroll manually to the start of the history
**Actual result:**  "Tribute paid" message is shown
**Video:** https://drive.google.com/open?id=1KxhDHWkruqATKLr3Q9rPR_9qSTeBRlSq

#### 10. "Confirmed" is shown on "Failed" transaction in chat (likely in nightly as well)
I don't have steps to reproduce, it just transaction for paying TTT is failed with "Out of gas" error - but in chat, I see that transaction is confirmed.
Transaction: https://ropsten.etherscan.io/tx/0x6429f6b29c306d5d62e977b64220678f746edaed20c82447b8f38e5ce48286d7
How it looks in chat and in transaction history accordingly:
photo_2019-06-03 12 24 02 jpeg 2019-06-03 12-24-39
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/3247071/Status-debug-logs.zip)

### on build 32 and 33:
#### 11. No preview on home for last messages in public and group chat
**Steps:**
- create account
- join any public chat with messages
- tap back
**Expected result:** can see the preview of the last message
**Actual result:** no preview
IMG_BD45AF7E37BC-1 jpeg 2019-06-03 12-32-27

**Comment by yenda** [comment]
@churik Let's focus on build without ttt from now on. Fixed the last message it was just a wrong condition for group-chats. Can you create issues for the ttt enabled buid? I will make a seperate PR enabling ttt and fixing these issues

**Comment by churik** [comment]
@yenda 
no preview for "No messages" state in last build (34)
IMG_0F3DC0EF6D1D-1

**Comment by yenda** [comment]
@churik damn, fixed

**Comment by churik** [comment]
@yenda one more:
error when when enable "Preview privacy mode" in Profile.
IMG_9CDEA0EF12B7-1

Logs: [status_logs1.zip](https://github.com/status-im/status-react/files/3247483/status_logs1.zip)

**Comment by yenda** [comment]
@churik no related to this PR

**Comment by yenda** [comment]
@churik fixed here https://github.com/status-im/status-react/pull/8345/files

**Comment by churik** [comment]
Checked Mac OS build (chatting) - no difference from my POV with current nightly.
Full regression from build 33: https://ethstatus.testrail.net/index.php?/runs/view/3505 (no regression found by e2e)
Checked IOS and Android on build 35:
- issue with preview is fixed
- no TTT in Profile
- chatting 
- sending transactions.

All issues when TTT is enabled will be added and fixed separately.

**Pull request merged**